server/src/main/java/org/opensearch/action/admin/cluster/shards/CatShardsResponse.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.shards;
+
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * A response of a cat shards request.
+ *
+ * @opensearch.api
+ */
+public class CatShardsResponse extends ActionResponse {
```

Comment:
Can we use a more generic name given this response contains botth cluster state and indices stats. Wondering if we can leverage this for indices stats and other calls

Reply:
We can give generic naming but as of today i don't think this response can be used by any other admin apis except Cat shards.

Reply:
You can create AbstractCatResponse which can have any common responses. I think ClusterStateResponse will be common across all cat APIs.

Reply:
Every admin apis have different set of responses, i don't think we can write common things.